MaintenanceDashboard.Data/API/EmployeeContext.cs

```diff
@@ -2,10 +2,11 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using MaintenanceDashboard.Data.Interfaces;
 
 namespace MaintenanceDashboard.Data.Api
 {
-    public class EmployeeContext : IDisposable, IEmployeeContext
```

Comment:
Doczytaj czym jest `IDisposable` i jak się tym posługiwać. Na pewno nie woła się w metodzie `Dispose` metod pochodzących z `GC`. Ogólnie zasada jest prosta, jeśli wołasz garbage collector w kodzie to albo wiesz PERFEKCYJNIE co robisz, albo nie wiesz totalnie nic ;p

Long story short, IDisposable głównie używa się do klas które pierwotnie zostały napisane w np. C++ (np. `HttpClient` i są tylko wywoływane przez runtime .NET. .NETowy GC nie umie czyścić pamięci niezarządzanej (jak sama nazwa wskazuje, że nią nie zarządza) dlatego MS wymyślił taki mechanizm by czyścić pamięć kiedy obiekt nie jest używany. Polecam doczytać, bo pytanie o IDisposable zdarza się na rekrutacjach.